### PR TITLE
remove pools and coingecko prices from tRPC

### PR DESCRIPTION
**Pull request by p0mvn**
- Removes pools prices
- Removes coingecko prices. Note that SQS auto-fallbacks to Coingecko
- Removes redundant parameters, simply providing denom to the getAssetPrice tRPC
- Overall, simplifies the tRPC, drastically reducing tech debt

### CODE REVIEW

**Review by jonator — CHANGES_REQUESTED** [review]

File: packages/web/server/api/routers/bridge-transfer.ts [hunk 1]
```diff
@@ -130,16 +130,7 @@ export const bridgeTransferRouter = createTRPCRouter({
...
         // since we're "bridging" the same variant, the to or from
         // asset could be used for pricing
-        getAssetPrice({
-          ...ctx,
-          asset: {
-            coinMinimalDenom: input.toAsset.address,
-            sourceDenom: input.toAsset.address,
-            chainId: input.toChain.chainId,
-            address: input.toAsset.address,
-            coinGeckoId: input.toAsset.coinGeckoId,
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply packages/web/server/api/routers/bridge-transfer.ts:140]: >CoinGeckoID: when there's an asset on another chain that's returned from a provider that we don't have represented in our asset list (or may not even have liquidity on Osmosis). One example of the top of my head is USDC on DYDX being used as an alt fee token. Yes, we could add that to the counterparty array for USDC on our chain, but this prevents @JeremyParish69 from having to register every possible permutation of assets on all other chains within our asset list to keep it lean, and allows us to adapt to new chains and/or chains adding new alt fee tokens dynamically.

Can we create a separate tRPC for CoinGecko prices separately and avoid coupling 2 distinct usecases?

Could you please also link the code where it is relevant?

File: packages/server/src/queries/complex/assets/price/index.ts [hunk 2]
```diff
@@ -1,92 +1,35 @@
...
-  currency = "usd",
-  priceProvider = getPriceFromSidecar,
-}: {
-  chainList: Chain[];
-  assetLists: AssetList[];
-  asset: { coinDenom?: string } & (
-    | { coinMinimalDenom: string }
-    | { sourceDenom: string }
-    | { chainId: number | string; address: string }
-    | { coinGeckoId: string }
```
[the remark this answers is not in the document]
Comment (p0mvn) [reply packages/server/src/queries/complex/assets/price/index.ts:39]: I think we should decouple the implementations for these usecases

**Review by p0mvn — COMMENTED** [review]
Thanks for the review @jonator . Left my thoughts

File: packages/web/server/api/routers/bridge-transfer.ts [hunk 3]
```diff
@@ -154,27 +145,9 @@ export const bridgeTransferRouter = createTRPCRouter({
...
-          return getAssetPrice({
-            ...ctx,
-            asset: {
-              coinMinimalDenom: input.fromAsset.address,
-              sourceDenom: input.fromAsset.address,
-              chainId: input.fromChain.chainId,
-              address: input.fromAsset.address,
-              coinGeckoId: input.fromAsset.coinGeckoId,
-            },
-          });
```
Comment (p0mvn) [inline packages/web/server/api/routers/bridge-transfer.ts:166]: 3 cases:
- `coindMinimalDenom`
- `chainID` & `address`
- `coinGeckoId`

**Comment by p0mvn** [comment]
To address Jon's feedback about the aggressive prices refactoring while not accounting for the bridging needs, I separated the work into incremental steps:
- Remove expensive pools prices source, R4R: https://github.com/osmosis-labs/osmosis-frontend/pull/3664
- Remove source denom (small api improvement) R4R: https://github.com/osmosis-labs/osmosis-frontend/pull/3665
- Implement getNativePrices function that calls SQS only and make it be used in all use cases other than bridging, R4R: https://github.com/osmosis-labs/osmosis-frontend/pull/3666

The main motivation for the change is simply calling SQS in 90% of the cases for simplicity and ease of identifying the performance bottlenecks.

Saw the prices tRPC sometimes taking ~700 ms consistently in the router when SQS takes never takes above 15 ms.

There are 3 pricing sources under the hood. I removed one (pools) - it is legacy, and we have no need for it anymore. Removed Coingecko fallback for all cases other than bridging assets where we might need to get prices from Coingecko IDs and some of the asset list chain counterparty parameters.

This will help with further digging deeper into the performance issues but also improve performance from not having to fallback on suboptimal pricing sources when price is requested for an unsupported denom.

This change also significantly improves readability by removing a number of redundant parameters.

Would appreciate a look and approval from FE devs. I will request the QA once the last PR is accepted.

**Pull request closed without merging**